### ISSUE TITLE
Problems should not dissapear on file close

### ISSUE THREAD

**Issue by leonbloy** [microsoft/vscode-eslint#1460]
Currently, when the extension has detected some problem in some file and it appears in the Problem panel, that problem "disappears" the moment the file is closed. 

This is highly undesirable - lacking some mechanism to "open all files with problems", that info is lost.

Worst still, this happens also is the file was not originally open, 

For example:
 - Enable the eslint task.
 - Add some typo to some foo.js file and close it (or edit it from outside VScode)
 - Run the eslint task - the error is found and appears in the Problem panel.
 - Open the file, from the Problem panel. And close it.
 - Result: the problem disappears.

It would be nice if the linter results are kept unless the file is modified.

**Comment by ruffin--** [comment]
2¢: If this is implemented, please make this optional/a setting. There are times I have legacy code where, if I had to correct everything in the project, would take years. ;) Being able to close files and concentrate on just what I'm working on today is live-saving.

**Comment by Mike-Logit** [comment]
I should have posted my feature request here instead of on the `vscode` main git repo but I wasn't sure if it was a VSCode issue or an issue with ESLint because other people have reported the same thing with typescript compiler warnings.
https://github.com/microsoft/vscode/issues/155172

**Comment by dbaeumer** [comment]
VS Code advice is for linters to only show errors for opened files (which the extension does). If we want to keep them then we need to turn ESLint into a workspace diagnostic provider which always lints all files (even if they never got opened) to conform to VS Code's diagnostic model.

**Comment by jgonzalezalbisu** [comment]
Hi @dbaeumer , can we currently do that?

**Comment by dbaeumer** [comment]
Let ESLint lint all files in a workspace. Yes, by using VS Code's task infrastructure but not via the extension itself. I am reluctant to implement this due to the performance implications it has.

**Comment by hilycker** [comment]
I agree with performance implications it could have, but it doesn't have to be turned on by default. Just let users turn it of via setting with maybe a note/warning about possible performance hits. Then it's user's responsibility.
